tests/factories.py

```diff
@@ -50,6 +51,62 @@ def ensure_workspace(workspace_or_name):
     raise Exception(f"Invalid workspace: {workspace_or_name})")  # pragma: nocover
 
 
+# get_output_metadata is imported from job-runner
+def get_output_metadata(
+    abspath, level, job_id, job_request, action, commit, excluded, message=None
+):
+    stat = abspath.stat()
+    with abspath.open("rb") as fp:
+        content_hash = file_digest(fp, "sha256").hexdigest()
+
+    return {
+        "level": level,
+        "job_id": job_id,
+        "job_request": job_request,
+        "action": action,
+        "commit": commit,
+        "size": stat.st_size,
+        "timestamp": stat.st_mtime,
+        "content_hash": content_hash,
+        "excluded": excluded,
+        "message": message,
+    }
```

Comment:
It also needs the row_count and col_count for csvs. In the version of this function that I wrote in parallel, I just had this extra bit of code (can add as part of my PR instead)
```
rows = cols = None
if abspath.suffix == ".csv":
    with abspath.open() as fp:
        reader = csv.DictReader(fp)
        first_row = next(reader, None)
        if first_row:
            cols = len(first_row)
            rows = sum(1 for _ in reader) + 1
        else:
            cols = rows = 0
```

Reply:
right